src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/CommandLineParser.cs

```diff
@@ -25,13 +25,14 @@ internal class CommandSwitch
 
         internal CommandSwitch(string name, string abbreviation, SwitchType switchType)
         {
-            //ensure that either name doesn't start with '/' or '-'
+            //ensure that either name doesn't start with '--' or '-'
             //also convert to lower-case
-            if ((name[0] == '/') || (name[0] == '-'))
-                _name = (name.Substring(1)).ToLower(CultureInfo.InvariantCulture);
+            if (name.Length > 1 && name[0] == '-' && name[1] == '-')
```

Comment:
Looks like this statement didn't handled "name" starts with "-"

Reply:
@Lxiamail This constructor is only called in OptionsStatics.cs line 52-62. All names and abbreviations are passed in correctly. So I feel the check and conversion is not necessary here. Anyway, this code is to remove the "--" before name and "-" before abbrevation.